pkg/controllers/provisioningv2/rke2/planner/controller.go

```diff
@@ -51,37 +52,56 @@ func Register(ctx context.Context, clients *wrangler.Context, planner *planner.P
 	}, clients.RKE.RKEControlPlane(), clients.Core.Secret(), clients.CAPI.Machine())
 }
 
-func (h *handler) OnChange(cluster *rkev1.RKEControlPlane, status rkev1.RKEControlPlaneStatus) (rkev1.RKEControlPlaneStatus, error) {
-	logrus.Debugf("[planner] rkecluster %s/%s: handler OnChange called", cluster.Namespace, cluster.Name)
-	if !cluster.DeletionTimestamp.IsZero() {
+func (h *handler) OnChange(cp *rkev1.RKEControlPlane, status rkev1.RKEControlPlaneStatus) (rkev1.RKEControlPlaneStatus, error) {
+	logrus.Debugf("[planner] rkecluster %s/%s: handler OnChange called", cp.Namespace, cp.Name)
+	if !cp.DeletionTimestamp.IsZero() {
 		return status, nil
 	}
 
-	status.ObservedGeneration = cluster.Generation
+	status.ObservedGeneration = cp.Generation
 
-	logrus.Debugf("[planner] rkecluster %s/%s: calling planner process", cluster.Namespace, cluster.Name)
-	err := h.planner.Process(cluster)
-	var errWaiting planner.ErrWaiting
-	if errors.As(err, &errWaiting) {
-		logrus.Infof("[planner] rkecluster %s/%s: waiting: %v", cluster.Namespace, cluster.Name, err)
-		rke2.Ready.SetStatus(&status, "Unknown")
-		rke2.Ready.Message(&status, err.Error())
-		rke2.Ready.Reason(&status, "Waiting")
-		return status, nil
+	if rke2.Reconciled.IsTrue(&status) && !equality.Semantic.DeepEqual(cp.Spec, status.AppliedSpec) {
+		rke2.Reconciled.SetStatus(&status, "Unknown")
+		rke2.Reconciled.Message(&status, "reconciling control plane")
+		rke2.Reconciled.Reason(&status, "Waiting")
 	}
-	if !errors.Is(err, generic.ErrSkip) {
-		rke2.Ready.SetError(&status, "", err)
-		if err != nil {
-			// don't return error because the controller will reset the status and then not assign the error
-			// because we don't register this handler with an associated condition. This is pretty much a bug in the
-			// framework but it's too impactful to change right before 2.6.0 so we should consider changing this later.
-			// If you are reading this years later we'll just assume we decided not to change the framework.
-			logrus.Errorf("[planner] rkecluster %s/%s: error encountered during plan processing was %v", cluster.Namespace, cluster.Name, err)
-			h.controlPlanes.EnqueueAfter(cluster.Namespace, cluster.Name, 5*time.Second)
+
+	logrus.Debugf("[planner] rkecluster %s/%s: calling planner process", cp.Namespace, cp.Name)
+	status, err := h.planner.Process(cp, status)
+	if err != nil {
+		if planner.IsErrWaiting(err) {
+			logrus.Infof("[planner] rkecluster %s/%s: waiting: %v", cp.Namespace, cp.Name, err)
+			rke2.Reconciled.Message(&status, "reconciling control plane")
```

Comment:
technically, if we're waiting we could be reconciling the worker nodes as well. this will lead to user confusion imo